src/azul/indexer/index_service.py

```diff
@@ -628,15 +634,15 @@ def _on_success(self, doc: Document):
         else:
             log.debug('Successfully wrote %s.', coordinates)
 
-    def _on_error(self, doc: Document, e):
+    def _on_error(self, doc: Document, e, exc_info=None):
         self.errors[doc.coordinates] += 1
         if self.error_retry_limit is None or self.errors[doc.coordinates] <= self.error_retry_limit:
             action = 'retrying'
             self.retries.add(doc.coordinates)
         else:
             action = 'giving up'
         log.warning('There was a general error with document %r: %r. Total # of errors: %i, %s.',
-                    doc.coordinates, e, self.errors[doc.coordinates], action)
+                    doc.coordinates, e, self.errors[doc.coordinates], action, exc_info=exc_info)
```

Comment:
```suggestion
                    doc.coordinates, e, self.errors[doc.coordinates], action, exc_info=True)
```

should be sufficient.

Reply:
Why do we need this?

Why doesn't _on_conflict have this, too?

Do we still need the `e` argument? I thought sys.exc_info() returns a reference to the currently handled exception.

Reply:
> Why do we need this?

When testing, I screwed something up and the name of the exception was printed, but not the stacktrace which made debugging more difficult. Adding this helped.

> Why doesn't _on_conflict have this, too?

It probably could, but we tend to know why _on_confict errors happen so it wouldn't add much debugging info.

> `exc_info=True` should be sufficient

It is, and makes passing `exc_info` unnecessary, *except*, It is not helpful when processing bulk. I have a simpler solution that should avoid these issues, but if you don't like it, maybe we should just drop the commit.